### PR TITLE
Add phone model

### PR DESCRIPTION
**Pull request by uberSaiyan**

### CODE REVIEW

File: src/main/java/seedu/address/model/phone/Capacity.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package seedu.address.model.phone;
+
+/**
+ * Represents a Phone's memory capacity in the SML.
+ */
+public enum Capacity {
+    SIZE_8GB,
```
Comment (EugeneTeu) [inline src/main/java/seedu/address/model/phone/Capacity.java]: do we need SIZE or just 8GB / 64GB etc ?

File: src/main/java/seedu/address/model/phone/Phone.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
+package seedu.address.model.phone;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.awt.Color;
```
Comment (EugeneTeu) [inline src/main/java/seedu/address/model/phone/Phone.java]: Do we want to use a custom class? Like might have custom colour not supposed by the built in e.g Midnight green

**Comment by EugeneTeu** [comment]
Generally okay, need to confirm the above things before merge

File: src/main/java/seedu/address/model/phone/Capacity.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    SIZE_16GB,
+    SIZE_32GB,
+    SIZE_64GB,
+    SIZE_128GB,
+    SIZE_256GB,
+    SIZE_512GB
```
Comment (qiujiaaa) [inline src/main/java/seedu/address/model/phone/Capacity.java]: can include 1tb too

File: src/main/java/seedu/address/model/phone/Capacity.java (hunk 1 above)
Comment (uberSaiyan) [reply src/main/java/seedu/address/model/phone/Capacity.java]: Enum constant identifiers can't start with numbers, so had no choice. But taking suggestions on what prefix to put.

Comment (EugeneTeu) [reply src/main/java/seedu/address/model/phone/Capacity.java]: ok I guess we have to keep it as SIZE_8GB

File: src/main/java/seedu/address/model/phone/Cost.java [hunk 4]
```diff
@@ -1,49 +1,58 @@
 package seedu.address.model.phone;
 
+import java.text.NumberFormat;
+import java.text.ParseException;
+
 import static seedu.address.commons.util.AppUtil.checkArgument;
```
Comment (EugeneTeu) [inline src/main/java/seedu/address/model/phone/Cost.java]: this import line violates check style

Comment (uberSaiyan) [reply src/main/java/seedu/address/model/phone/Cost.java]: thx, changed

**Pull request merged**